kernel/src/utils.rs

```diff
@@ -21,7 +22,11 @@ pub(crate) mod test_utils {
     use tempfile::TempDir;
     use test_utils::delta_path_for_version;
 
-    use crate::actions::{Add, Cdc, CommitInfo, Metadata, Protocol, Remove};
+    use crate::{
+        actions::{Add, Cdc, CommitInfo, Metadata, Protocol, Remove},
+        engine::arrow_data::ArrowEngineData,
+        EngineData,
+    };
```

Comment:
separate `use` statements is easier to read and usually fewer lines as well:
```suggestion
    use crate::actions::{Add, Cdc, CommitInfo, Metadata, Protocol, Remove};
    use crate::{engine::arrow_data::ArrowEngineData, EngineData};
```
or even 
```suggestion
    use crate::actions::{Add, Cdc, CommitInfo, Metadata, Protocol, Remove};
    use crate::engine::arrow_data::ArrowEngineData;
    use crate::EngineData;
```